code/src/components/WelcomeScreen.js

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import game, { generateGame } from "../reducers/game";
+
+export const WelcomeScreen = () => {
+  const userName = useSelector((store) => store.game.userName);
+  const dispatch = useDispatch();
+
+  const onGameStartSet = (event) => {
+    event.preventDefault();
+    dispatch(generateGame(userName));
+    dispatch(game.actions.setGameStart(true));
+  };
+
+  return (
+    <>
```

Comment:
You have only one immediate child element (div) so you shouldn't use react fragment here.